strongbox-security/strongbox-security-api/src/test/java/org/carlspring/strongbox/configuration/StrongboxDelegatingPasswordEncoderTest.java

```diff
@@ -0,0 +1,53 @@
+package org.carlspring.strongbox.configuration;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.util.Base64Utils;
+
+
+@SpringBootTest
+@ActiveProfiles(profiles = "test")
+@ContextConfiguration
+class StrongboxDelegatingPasswordEncoderTest
+{
+
+    @org.springframework.context.annotation.Configuration
+    @ComponentScan(basePackages = { "org.carlspring.strongbox.security",
+                                    "org.carlspring.strongbox.testing",
+                                    "org.carlspring.strongbox.configuration" })
+    public static class SpringConfig
+    {
+
+    }
+
+
+    @Inject
+    private PasswordEncoder passwordEncoder;
+
+    @Test
+    void testNullEncodeAndMatch()
+    {
+        Assertions.assertThrows(NullPointerException.class, () -> passwordEncoder.encode(null));
+        Assertions.assertTrue(passwordEncoder.matches(null, null));
+    }
+
+    @Test
+    void encodeAndMatch()
```

Comment:
This test case is not properly testing the real scenario. Check #1840 (specifically the `{alg}` part).
If it is supposed to test a raw password which is not hashed and has no hash algorithm - the method should probably be renamed to something like `rawNoAlgPasswordShouldEncode()`. 
Naming is important as it gives context to what this test is all about!